numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -1581,7 +1573,10 @@ gentype_byteswap(PyObject *self, PyObject *args, PyObject *kwds)
         PyObject *new;
         char *newmem;
 
-        gentype_getreadbuf(self, 0, (void **)&data);
+        PyArray_Descr *outcode = PyArray_DescrFromScalar(self);
+        data = (void *)scalar_value(self, outcode);
+        Py_DECREF(outcode);
+
         descr = PyArray_DescrFromScalar(self);
```

Comment:
This line is repeated now, maybe we can merge them

Reply:
I'm less comfortable in python c code, and I consolidated two Py_DECREF to one place which felt better but made me question why they were seperated before. Can you give my update a through inspection.

Reply:
You missed that `descr` is still being used further down in `new = ...`

Reply:
descr is a better name than outcode anyway

Reply:
> You missed that `descr` is still being used further down in `new = ...`

I'm not sure what you're implying by this? I dropped `outcode` so I think desc should still be used in the `new = ...`.

Reply:
Yes, but as long as you are using `descr` you must hold a reference to it. If you call `DECREF` you lose that reference (i.e. if you wanted to be super clean, at least in this case, you might even do `descr = NULL` or similar incarnations to point out that it is not anymore valid to use).

Reply:
Thanks, I was being silly and only making sure the count was right at the end of the function. Fixed now.